core/src/com/unciv/ui/tilegroups/CityButton.kt

```diff
@@ -155,6 +159,52 @@ class CityButton(val city: CityInfo, internal val tileGroup: WorldTileGroup, ski
         parent.addAction(floatAction)
     }
 
+    private fun getPopulationGroup(showGrowth: Boolean): Group {
+        val group = Group()
+
+        val populationLabel = city.population.population.toString().toLabel()
+
+        group.addActor(populationLabel)
+
+        val groupHeight = 25f
+        var groupWidth = populationLabel.width
+        if (showGrowth) groupWidth += 10f
+        group.setSize(groupWidth, groupHeight)
+
+        if (showGrowth) {
+            var growthPercentage = city.population.foodStored / city.population.getFoodToNextPopulation().toFloat()
+            if (growthPercentage < 0) growthPercentage = 0.0f
+
+            val growthBar = ImageGetter.getProgressBarVertical(2f, groupHeight, growthPercentage,
+                    if (city.isStarving()) Color.RED else Color.GREEN, Color.BLACK)
+            growthBar.x = populationLabel.width + 3
+            growthBar.centerY(group)
+
+            group.addActor(growthBar)
+
+            val turnLabel : Label
+            if (city.isGrowing()) {
+                turnLabel = city.getNumTurnsToNewPopulation().toString().toLabel()
+                turnLabel.color = Color.GREEN
+            } else if (city.isStarving()) {
+                turnLabel = (-city.getNumTurnsToStarvation()).toString().toLabel()
+                turnLabel.color = Color.RED
+            } else {
+                turnLabel = "-".toLabel()
+                turnLabel.color = Color.WHITE
```

Comment:
I think there should some upper bound on turns to growth and starvation. Probably if it's higher than 99 you can show an ∞ symbol? 
Also in case the population is stable maybe show ∞? (or don't show at all the growth group?)

In case of starvation why is the number of turns shown as negative?

Reply:
I have tried to display the infinity symbol, but for some reason it doesn't show it in-game. Maybe the graphics library isn't configured for unicode properly?

Reply:
To display a unicode symbol you have to add first to the set of symbols loaded: https://github.com/yairm210/Unciv/blob/c6b96f4f5b711aab3a77a249767b522775e312bd/core/src/com/unciv/ui/utils/Fonts.kt#L21